android/src/main/java/com/vinzscam/rntusclient/RNTusClientModule.java

```diff
@@ -99,39 +99,38 @@ public void resume(String uploadId, Callback callback) {
   public void abort(String uploadId, Callback callback) {
     try {
       TusRunnable executor = this.executorsMap.get(uploadId);
-      if(executor != null) {
+      if (executor != null) {
         executor.finish();
       }
-      callback.invoke((Object)null);
-    } catch(IOException | ProtocolException e) {
+      callback.invoke((Object) null);
+    } catch (IOException | ProtocolException e) {
       callback.invoke(e);
     }
   }
 
   class TusRunnable extends TusExecutor implements Runnable {
-    private TusUpload upload;
+    private TusAndroidUpload upload;
     private TusUploader uploader;
     private String uploadId;
     private TusClient client;
     private boolean shouldFinish;
     private boolean isRunning;
 
-    public TusRunnable(String fileUrl,
-                       String uploadId,
-                       String endpoint,
-                       Map<String, String> metadata,
-                       Map<String, String>headers
-    ) throws FileNotFoundException, MalformedURLException {
+    public TusRunnable(String fileUrl, String uploadId, String endpoint, Map<String, String> metadata,
+        Map<String, String> headers) throws FileNotFoundException, MalformedURLException {
       this.uploadId = uploadId;
+
       client = new TusClient();
       client.setUploadCreationURL(new URL(endpoint));
 
-      SharedPreferences pref = getReactApplicationContext().getSharedPreferences("tus", 0);
+      SharedPreferences pref = reactContext.getSharedPreferences("tus", 0);
+
       client.enableResuming(new TusPreferencesURLStore(pref));
       client.setHeaders(headers);
-      File file = new File(fileUrl);
-      upload = new TusUpload((file));
+
+      upload = new TusAndroidUpload(Uri.parse(fileUrl), reactContext);
```

Comment:
Isn't the second parameter supposed to be of type "Activity" as per the following documentation https://tus.github.io/tus-android-client/javadoc/io/tus/android/client/TusAndroidUpload.html ?

Reply:
Yes! I tried with getCurrentActivity() too, but for some reason it was not working properly for me (Xiaomi Mi 9). Then I tried with reactContext instead and it worked great.

Reply:
The result is currently in production btw:
https://play.google.com/store/apps/details?id=io.starchive.mobile